internal/envoy/route.go

```diff
@@ -116,6 +124,18 @@ func mirrorPolicy(r *dag.Route) *envoy_api_v2_route.RouteAction_RequestMirrorPol
 	}
 }
 
+func hostReplaceHeader(hp *dag.RequestHeadersPolicy) string {
+	if hp == nil {
+		return ""
+	}
+	for _, rp := range hp.Set {
+		if strings.EqualFold(rp.Key, "Host") {
```

Comment:
Sure, but this is a special use-case. I don't see the value of reformatting a specific special string that we're looking for.

Reply:
It's laying down the APIs and infrastructure for the general case. We can make the change later, but to me it seems that we might as well do it now since we have a good idea of where we want this code to end up.